controllers/release/adapter.go

```diff
@@ -246,36 +246,33 @@ func (a *adapter) EnsureReleaseIsProcessed() (controller.OperationResult, error)
 	return controller.ContinueProcessing()
 }
 
-// EnsureReleaseIsValid is an operation that will ensure that a Release is valid by checking all the resources needed
-// to process it.
+// EnsureReleaseIsValid is an operation that will ensure that a Release is valid by performing all
+// validation checks.
 func (a *adapter) EnsureReleaseIsValid() (controller.OperationResult, error) {
 	patch := client.MergeFrom(a.release.DeepCopy())
-	resources, err := a.loader.GetProcessingResources(a.ctx, a.client, a.release)
-
-	if err != nil {
-		if resources == nil || resources.ReleasePlanAdmission == nil || errors.IsNotFound(err) {
-			a.release.MarkValidationFailed(err.Error())
-			a.release.MarkReleaseFailed("Release validation failed")
-			return controller.RequeueOnErrorOrStop(a.client.Status().Patch(a.ctx, a.release, patch))
-		}
 
-		return controller.RequeueWithError(err)
+	validationFuncs := []func() (bool, error){
+		a.validateAuthor,
+		a.validateProcessingResources,
 	}
 
-	err = a.validateAuthor(resources.ReleasePlan)
-	if err != nil {
-		if strings.Contains(err.Error(), "automated not set in status") {
+	for _, validationFunc := range validationFuncs {
+		valid, err := validationFunc()
+		if err != nil {
 			return controller.RequeueWithError(err)
 		}
-
-		a.release.MarkValidationFailed(err.Error())
-		a.release.MarkReleaseFailed("Author validation failed")
-		return controller.RequeueOnErrorOrStop(a.client.Status().Patch(a.ctx, a.release, patch))
+		if !valid {
+			a.release.MarkReleaseFailed("Release validation failed")
```

Comment:
Should we `break` here? No point on continuing.

Reply:
fixed